### DISCUSSION TITLE
Replace JSON serializer with System.Text.Json?

### DISCUSSION THREAD

**Discussion by michaeljon** [quartznet/quartznet#1378] [Q&A]
Is it possible to swap out the JSON.Net serializer with System.Text.Json? We have a large number of custom serializers that define our query language(s) and moving them to JSON.Net would essentially be a product rewrite for us. We could just store the query text as a raw string but that just seems like the wrong approach. -- thanks

**Comment by lahma** [comment]
If you have no existing data in database and there's no migration concern, sure. If you implement `IObjectSerializer` that does `System.Text.Json` serialization that would handle the case. You can check the current Newtonsoft.JSON-based version for reference.

Then it would be just case of configuring via `persistentStoreOptions.UseSerializer<MyCustomSerializer>()` or in configuration file / properties with `quartz.serializer.type=MyCustomSerializer, MyAssembly`

A new project supporting such serialization would be a great and welcome addition as a PR, but it should also support deserializing existing data that was persisted with Json.NET (`$type` etc).

**Reply by michaeljon** [reply]
I'll have a look. If there are any changes necessary to the core libs I'll address and send you a PR.

**Reply by lahma** [reply]
As Quartz has no dependencies to specific libraries it would require a separate project/NuGet package.
